textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcherTest.java

```diff
@@ -6,78 +6,163 @@
 import org.junit.Assert;
 import org.junit.Test;
 
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IField;
 import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.common.Schema;
+import edu.uci.ics.textdb.common.constants.SchemaConstants;
 import edu.uci.ics.textdb.common.constants.TestConstants;
+import edu.uci.ics.textdb.common.field.DataTuple;
+import edu.uci.ics.textdb.common.field.ListField;
+import edu.uci.ics.textdb.common.field.Span;
 
 /**
  * @author zuozhi
- * @author laisycs
+ * @author shuying
  * @author chenli
  * 	
  * Unit test for RegexMatcher
  */
 public class RegexMatcherTest {
+	private Schema createSpanSchema(Schema schema) {
+    	List<Attribute> attributesCopy = new ArrayList<>(schema.getAttributes());
+    	attributesCopy.add(SchemaConstants.SPAN_LIST_ATTRIBUTE);
+    	return new Schema(attributesCopy.toArray(new Attribute[attributesCopy.size()]));
+    }
+	
 	@Test
 	public void testGetNextTuplePeopleFirstName() throws Exception {
 		List<ITuple> data = TestConstants.getSamplePeopleTuples();
-		RegexMatcherTestHelper test = new RegexMatcherTestHelper(TestConstants.SCHEMA_PEOPLE, data);
+		RegexMatcherTestHelper testHelper = new RegexMatcherTestHelper(TestConstants.SCHEMA_PEOPLE, data);
+		
+		testHelper.runTest("g[^\\s]*", TestConstants.FIRST_NAME);
 		
 		List<ITuple> expected = new ArrayList<ITuple>();
-		expected.add(data.get(0));
-		expected.add(data.get(2));
-		test.runTest("b.*", TestConstants.FIRST_NAME);
-		Assert.assertTrue(test.matchExpectedResults(expected));
 		
-		test.cleanUp();
+		//expected to match "brad lie angelina"
+		Schema spanSchema = createSpanSchema(TestConstants.SCHEMA_PEOPLE);
+		List<Span> spans = new ArrayList<Span>();
+		spans.add(new Span(TestConstants.FIRST_NAME, 11, 17, "g[^\\s]*", "brad lie angelina"));
+		IField spanField = new ListField<Span>(new ArrayList<Span>(spans));
+		List<IField> fields = new ArrayList<IField>(data.get(2).getFields());
+		fields.add(spanField);
+		expected.add(new DataTuple(spanSchema, fields.toArray(new IField[fields.size()])));
+		
+		//expected to match "george lin lin"
+		spans.clear();
+		spans.add(new Span(TestConstants.FIRST_NAME, 0, 6, "g[^\\s]*", "george lin lin"));
+		spanField = new ListField<Span>(new ArrayList<Span>(spans));
+		fields = new ArrayList<IField>(data.get(3).getFields());
+		fields.add(spanField);
+		expected.add(new DataTuple(spanSchema, fields.toArray(new IField[fields.size()])));
+		
+		Assert.assertTrue(testHelper.matchExpectedResults(expected));
```

Comment:
Please use TestUtils.containsAllResults(). Do it for your test cases.  See the test cases of DictionaryMatcher for an example. Added @rajesh9625 to this discussion.